OpenRA.Mods.AS/Graphics/TintedCell.cs

```diff
@@ -0,0 +1,105 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2015- OpenRA.Mods.AS Developers (see AUTHORS)
+ * This file is a part of a third-party plugin for OpenRA, which is
+ * free software. It is made available to you under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation. For more information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Effects;
+using OpenRA.Graphics;
+using OpenRA.Mods.AS.Traits;
+
+namespace OpenRA.Mods.AS.Graphics
+{
+	public class TintedCell : IRenderable, IFinalizedRenderable, IEffect
+	{
+		public int Ticks = 0;
+		public int Level = 0;
+		readonly TintedCellsLayer layer;
+		readonly WPos wpos;
+
+		public int ZOffset
+		{
+			get
+			{
+				return layer.Info.ZOffset;
+			}
+		}
+
+		public TintedCell(TintedCellsLayer layer, WPos wpos)
+		{
+			this.wpos = wpos;
+			this.layer = layer;
+		}
+
+		public TintedCell(TintedCell src)
+		{
+			Ticks = src.Ticks;
+			Level = src.Level;
+			layer = src.layer;
+			wpos = src.wpos;
+		}
+
+		public IRenderable WithPalette(PaletteReference newPalette) { return this; }
+		public IRenderable WithZOffset(int newOffset) { return this; }
+		public IRenderable OffsetBy(WVec vec) { return this; }
+		public IRenderable AsDecoration() { return this; }
+
+		public PaletteReference Palette { get { return null; } }
+		public bool IsDecoration { get { return false; } }
+
+		public WPos Pos { get { return wpos; } }
+
+		IFinalizedRenderable IRenderable.PrepareRender(WorldRenderer wr)
+		{ return this; }
```

Comment:
Bad formatting - either move it into one row, either separate it to a full block.